### DISCUSSION TITLE
Retrieve data from survey-html-form

### DISCUSSION THREAD

**Discussion by christinalubinus** [jspsych/jsPsych#1260] [Q&A] [answered]
Hello everyone, 

In an attempt to build a rudimentary version of a digit span test, I plan to present participants with an audio file and ask them to reproduce the digits by typing them into a survey-html-form. To check their performance, I need to compare the digits typed in by the participant with the ones stored in a js-file. Thus, what I hope to get store in the variable _get_digits_ is just the digits that were typed in.
But... I am running into trouble already when extracting the results from the survey text field (I pasted the relevant -I hope- snippet of code below). My logic was that my variable _data_ would be an object, from which I could extract the value (the numbers typed in) by means of the key ("digits"). However, _get_digits_ is undefined and _data_ looks like this in the console:  ["{"digits":"456"}"]. I suppose I am just getting the data types wrong but I have tried a couple of things and just could not quite understand the logic behind it. Is there anyone who could help? Any hint would be appreciated. 

Thanks a lot in advance, 
Christina 

![Bildschirmfoto 2020-12-02 um 19 05 50](https://user-images.githubusercontent.com/45943148/100913350-f51c1680-34d1-11eb-8c21-9950a6c28c80.png)

**Comment by wenbo-lin** [comment]
Here is a quick example. 

```
var form_trial = {
		type: 'survey-html-form',
		preamble: '',
		html:'<input type="text" name="digits">',
		on_finish:function(data){
		let response = JSON.parse(data.responses);
		console.log(response.digits);
		}
	}
```
The input name portion is missing quotes around it.

**Reply by christinalubinus** [reply]
Thanks a lot for your quick reply! This helps a lot.

**Comment by becky-gilbert** [comment, marked as answer]
Hi Christina,
The problem here is that the value of responses starts off as an object of key-value pairs (question names and text responses), and JavaScript objects can't be stored directly in the jsPsych data, so they have to be stored as a JSON-formatted text. So when trying to access the responses object, you just need to convert the JSON text back to JavaScript with `JSON.parse()`. Try this:
```javascript
var form_trial = {
    type: 'survey-html-form',
    preamble: '<p>Please, type in the digits you just heard in the correct order.</p>',
    html: '<p><input name="digits" type="text"/></p>',
    button_label: 'Submit',
    on_finish: function(data) {
        var data = JSON.parse(jsPsych.data.get().last(1).select('responses').values);
        var get_digits = data["digits"];
        console.log(get_digits);
    }
}
```

**Reply by christinalubinus** [reply]
Thank you, Becky, the code solves the issue. Thanks as well for the explanation - the JSON-formatting information I did not quite absorb from the documentation (although it is there as I just checked).

**Reply by becky-gilbert** [reply]
Glad to hear that this is in the documentation, although I'm sure it could be clearer... We'll try to think of ways to improve it. (And by the way - if you ever want to help improve the jsPsych docs, you can always [submit a pull request](https://www.jspsych.org/about/contributing/#steps-for-modifying-the-code) with suggestions 😃 )

**Reply by christinalubinus** [reply]
Honestly, I have been wanting to start contributing to projects but didn't so far because I have not felt familiar enough with github. Maybe I take this hint as a challenge for the weekend 😉
